example-plugins/sethome.lua

```diff
@@ -0,0 +1,94 @@
+local lukkit = require("lukkit")
+
+local homestore_opts = {
+  store_type: lukkit.store.type.JSON,
+  overwrite_oncorrupt: false
+}
+
+
+-- Runs when the plugin is enabled
+function lukkit.plugin:enable()
+  local plugin_name = self.info:get_name();
+  self.logger:info("Enabling {}", plugin_name)
+end
+
+-- Runs when the plugin is disabled (shutdown, reload etc.)
+function lukkit.plugin:disable()
+  local plugin_name = self.info:get_name();
+  self.logger:info("Disabling {}", plugin_name)
+end
+
+-- Runs when the plugin is loaded (before enabled, on file read, can stop loading)
+function lukkit.plugin:load(load_context)
+  -- Can load a SQLite DB, JSON doc or YAML
+  local home_storage = self.store:load("home", homestore_opts)
+
+  -- do setup with store
+
+  if (home_storage:is_read_only()) then
+    -- Stop the plugin from continuining with a load
+    load_context:reject("Home store cannot be written to")
+  end
+end
+
+
+function lukkit.plugin:command(cmd, sender, args)
+  -- Handles any case permutation of "home" (e.g. "home", "Home", "HOME" etc.)
+  if (cmd:equals_lower("home")) then
+
+    -- Functional wrapper to only execute if the sender is a player, no-op if not
+    sender:if_player(function(player)
```

Comment:
DISC: functional usages

Reply:
Question: do you want to go all the way with functional programming or are you just satisfied with pulling only some features?

Reply:
I'm not a big fan of nulls/nils which is why I thought of pulling `Option` types into the language. I suppose it makes harder for an inexperienced dev which contradicts my previous comment so I might sleep on this one.

Reply:
As for this `if_player` instance there could also be an `is_player` which gives the user choice. Might be too much in the way of choice though. Thoughts?

Reply:
In this particular case, I believe `is_player` would be more flexible and better than having specialized functions with callbacks. With that you can use `if...else...end` and that feels more natural

Reply:
Gotcha! I think I went too far down the rabbit hole and `is_player` makes more sense from a newbie POV.